### PR TITLE
[SEM 458]: searchable filter control dropdown

### PR DESCRIPTION
**Pull request by mustartt**
Filterable FilterControl Dropdown

Screenshot from 2021-07-08 20-09-49

### CODE REVIEW

File: src/framework/filterControl/forms/SearchableMultipleSelectFilterForm.jsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        </div>
+      </div>
+    );
+  }
+}
+
+SearchableMultipleSelectFilterForm.propTypes = {
+  comparisonValue: PropTypes.array,
+  filterOption: PropTypes.instanceOf(FilterOption),
+  onAddFilter: PropTypes.func.isRequired,
```
Comment (el-chuck) [inline src/framework/filterControl/forms/SearchableMultipleSelectFilterForm.jsx:106]: should isImmediate be added?

File: src/framework/filterControl/forms/_searchableMultiSelectFilterForm.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.fitlerForm__search {
```
Comment (el-chuck) [inline src/framework/filterControl/forms/_searchableMultiSelectFilterForm.scss]: typo

File: src/framework/filterControl/forms/SearchableMultiSelectFilterForm.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
+import { TestCaseFactory } from 'react-test-kit';
```
Comment (el-chuck) [inline src/framework/filterControl/forms/SearchableMultiSelectFilterForm.spec.jsx]: filename should be SearchableMultipleSelect...

File: src/framework/filterControl/forms/FilterForm.jsx [hunk 4]
```diff
@@ -13,6 +13,8 @@ import DateRangeFilterForm from './DateRangeFilterForm.jsx';
 import InputFilterForm from './InputFilterForm.jsx';
 import MixedTypeValueFilterForm from './MixedTypeValueFilterForm.jsx';
 import MultipleSelectFilterForm from './MultipleSelectFilterForm.jsx';
+import SearchableMultipleSelectFilterForm
```
Comment (el-chuck) [inline src/framework/filterControl/forms/FilterForm.jsx:16]: why new line here?

File: src/framework/filterControl/forms/SearchableMultiSelectFilterForm.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+  });
+
+  describe('initial state', () => {
+    it('has options unselected', () => {
+      const props = {
+        filterOption,
+        onAddFilter: () => undefined,
+      };
+
+      const testCase =
```
Comment (el-chuck) [inline src/framework/filterControl/forms/SearchableMultiSelectFilterForm.spec.jsx]: also why new line here and below?

File: src/framework/filterControl/forms/SearchableMultiSelectFilterForm.spec.jsx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+
+      expect(
+        testCase.find('button')[0].className
+      ).toContain('is-button-disabled');
+    });
+  });
+
+  describe('Props', () => {
+    describe('filterOption', () => {
+      it('options are rendered', () => {
```
Comment (el-chuck) [inline src/framework/filterControl/forms/SearchableMultiSelectFilterForm.spec.jsx]: plus a test for search functionality

File: src/framework/filterControl/forms/FilterForm.jsx (hunk 4 above)
Comment (mustartt) [reply src/framework/filterControl/forms/FilterForm.jsx:16]: the eslint config has a strict 80 char limit and this line goes over that

File: src/framework/filterControl/forms/SearchableMultiSelectFilterForm.spec.jsx (hunk 5 above)
Comment (mustartt) [reply src/framework/filterControl/forms/SearchableMultiSelectFilterForm.spec.jsx]: the eslint config has a strict 80 char limit and this line goes over that

**Pull request merged**